lib/yi_appium_caps_util.rb

```diff
@@ -104,16 +104,25 @@ def update_ios_caps (output_data)
         %x[unzip -o #{myZipApp} -d #{myAppFolder}]
 
         puts "Launching getIP app"
-        install_app = %x[ios-deploy --justlaunch --bundle #{myApp}]
-        #Putting log into file
-        iplog = %x[(idevicesyslog) & sleep 5 ; kill $!]
-        File.write('iplog.txt', iplog)
-        #Getting ip from file
-        ip = %x[grep -m1 'IPAddress:' iplog.txt | awk '{print $8}']
+        #Launching app and putting (5 seconds of) log into file
+        ipcounter = 1
+        while ipcounter <= 5 do
+          puts "Try \# #{ipcounter}"
+          iplog = %x[ios-deploy --justlaunch --bundle #{myApp} & (idevicesyslog) & sleep 5 ; kill $!]
+          File.write('iplog.txt', iplog)
+          #Getting ip from file
+          ip = %x[grep -m1 'IPAddress:' iplog.txt | awk '{print $8}']
+          puts 'IP Address: ' + ip
+          if (ip == "")
+            ipcounter +=1
```

Comment:
Do you need to put a `wait` here to wait for a second if you want to keep trying for 5 seconds?